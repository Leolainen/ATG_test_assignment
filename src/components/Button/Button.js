import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./styles.module.css";

const Button = (props) => {
  const { className, children, ...other } = props;

  return (
    <button
      className={classnames(styles.root, styles.color, className)}
      data-testid="button"
      {...other}
    >
      <span className={styles.label}>{children}</span>
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

Button.uiName = "Button";

export default Button;
