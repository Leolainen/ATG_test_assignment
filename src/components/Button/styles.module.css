.root {
  cursor: pointer;
  padding: 0;
  margin: 0;
  outline: 0;
  border: none;
  appearance: none;
  position: relative;
  text-decoration: none;
  transition: all 125ms ease-out;
  display: inline-block;
  border-radius: 3px;
  padding: 4px 16px;
}

.color {
  background-color: var(--harmony-5);
}

.color:hover {
  background-color: var(--harmony-6);
}

.label {
  width: 100%;
  display: inherit;
  align-items: inherit;
  justify-content: inherit;
}
