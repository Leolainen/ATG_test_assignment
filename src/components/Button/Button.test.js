import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import Button from "./Button";

describe("<Button />", () => {
  afterEach(cleanup);

  describe("events", () => {
    it("should fire onClick callback when clicked", () => {
      const mockFn = jest.fn();
      const { getByTestId } = render(<Button onClick={mockFn}>button</Button>);
      fireEvent.click(getByTestId("button"));

      expect(mockFn).toHaveBeenCalled();
    });
  });

  describe("Snapshots", () => {
    it("should render", () => {
      const { asFragment } = render(<Button>button</Button>);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
