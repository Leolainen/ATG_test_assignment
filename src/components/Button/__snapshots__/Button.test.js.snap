// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots should render 1`] = `
<DocumentFragment>
  <button
    class="root color"
    data-testid="button"
  >
    <span
      class="label"
    >
      button
    </span>
  </button>
</DocumentFragment>
`;
